subprojects/tooling-api/src/main/java/org/gradle/tooling/model/idea/IdeaContentRoot.java

```diff
@@ -58,6 +59,40 @@
     @Incubating
     DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedTestDirectories();
 
+    /**
+     * The set of resource directories.
+     * NOTE: The resources directory is only available for Java projects, otherwise it is empty set.
+     * @throws UnsupportedMethodException For Gradle versions older than 4.7 where this method is not supported.
+     * @since 4.7
+     */
+    @Incubating
+    DomainObjectSet<? extends IdeaSourceDirectory> getResourceDirectories() throws UnsupportedMethodException;
+
+    /**
+     * The set of generated resource directories. This is a subset of those directories returned by {@link #getResourceDirectories()}.
+     * @throws UnsupportedMethodException For Gradle versions older than 4.7 where this method is not supported.
+     * @since 4.7
+     */
+    @Incubating
+    DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedResourceDirectories() throws UnsupportedMethodException;
+
+    /**
+     * The set of test resource directories.
+     * NOTE: The test resources directory is only available for Java projects, otherwise it is empty set.
+     * @throws UnsupportedMethodException For Gradle versions older than 4.7 where this method is not supported.
+     * @since 4.7
+     */
+    @Incubating
+    DomainObjectSet<? extends IdeaSourceDirectory> getTestResourceDirectories() throws UnsupportedMethodException;
+
+    /**
+     * The set of generated test resource directories. This is a subset of those directories returned by {@link #getTestResourceDirectories()}.
+     * @throws UnsupportedMethodException For Gradle versions older than 4.7 where this method is not supported.
+     * @since 4.7
+     */
+    @Incubating
+    DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedTestResourceDirectories() throws UnsupportedMethodException;;
```

Comment:
I don't know why we started this weird pattern of having convenience methods to get the generated dirs, but I don't think we should continue it. I'd probably deprecate the existing ones. The Tooling API should have as little surface as we can get away with.

Reply:
I have removed my two new getGenerated* methods and marked the remaining two as deprecated.